src/kmedoids_ucb.cpp

```diff
@@ -35,6 +35,15 @@ KMedoids::KMedoids(int n_medoids, std::string algorithm, int verbosity,
   KMedoids::checkAlgorithm(algorithm);
 }
 
+KMedoids::KMedoids(std::string algorithm, int verbosity,
```

Comment:
We shouldn't duplicate code: https://en.wikipedia.org/wiki/Don%27t_repeat_yourself

Instead, could you pass a default argument of null to the original function? Then, inside the function surface an error if that argument is still null.